impl/sql/sqlhist/sqlhist.go

```diff
@@ -260,23 +267,16 @@ func (m *Map) ReadLeafAt(txn transaction.Txn, index []byte, epoch int64) ([]byte
 }
 
 // NeighborsAt returns the list of neighbors from the neighbor leaf to just below the root at epoch.
-func (m *Map) NeighborsAt(ctx context.Context, index []byte, epoch int64) ([][]byte, error) {
-	tx, err := m.db.Begin()
+func (m *Map) NeighborsAt(txn transaction.Txn, index []byte, epoch int64) ([][]byte, error) {
+	nbrs, err := m.neighborsAt(txn, index, maxDepth, epoch)
 	if err != nil {
 		return nil, err
 	}
-	nbrs, err := m.neighborsAt(tx, index, maxDepth, epoch)
-	if err != nil {
-		if rbErr := tx.Rollback(); rbErr != nil {
-			err = fmt.Errorf("neighborsAt failed: %v, and Rollback failed: %v", err, rbErr)
-		}
-		return nil, err
-	}
 	cNbrs := compressNeighbors(m.mapID, nbrs, index, maxDepth)
-	return cNbrs, tx.Commit()
+	return cNbrs, nil
 }
 
-func (m *Map) neighborsAt(tx *sql.Tx, index []byte, depth int, epoch int64) ([]sparse.Hash, error) {
+func (m *Map) neighborsAt(tx transaction.Txn, index []byte, depth int, epoch int64) ([]sparse.Hash, error) {
```

Comment:
(Optional.) For consistency, we're using `txn` for transaction objects instead of `tx`.